tests/main/system-usernames-microk8s/task.yaml

```diff
@@ -0,0 +1,83 @@
+summary: ensure only approved snaps can use the microk8s system user
+
+# - not running on 14.04 as we have no real systemd here
+# - also exclude debian 10 because of old libseccomp (the system-usernames test
+#   is already checking which distributions have the needed support, so there's
+#   no need to replicate that code here)
+systems: [-ubuntu-14.04-*, -debian-10-*]
+
+environment:
+    STORE_DIR: $(pwd)/fake-store-blobdir
+
+prepare: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    echo "Ensure jq is installed"
+    if ! command -v jq; then
+        snap install --devmode jq
+    fi
+
+    echo "Ensure yaml2json is installed"
+    if ! command -v yaml2json; then
+        snap install --devmode remarshal
+    fi
+
+    snap debug can-manage-refreshes | MATCH false
+
+    snap ack "$TESTSLIB/assertions/testrootorg-store.account-key"
+
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+    setup_fake_store "$STORE_DIR"
+
+    cp "$TESTSLIB"/assertions/developer1.account "$STORE_DIR/asserts"
+    cp "$TESTSLIB"/assertions/developer1.account-key "$STORE_DIR/asserts"
+
+    snap ack "$TESTSLIB/assertions/developer1.account"
+    snap ack "$TESTSLIB/assertions/developer1.account-key"
+
+    # Create a snap which is not entitled to use the microk8s user
+
+    sed "snap/meta/snap.yaml.in" -e "s/SNAPNAME/test-microk8s-username/" > "snap/meta/snap.yaml"
+    snap_path=$("$TESTSTOOLS"/snaps-state pack-local snap)
+    make_snap_installable "$STORE_DIR" "${snap_path}"
```

Comment:
as per https://github.com/snapcore/snapd/pull/10494, you will need to ack the assertion here manually or otherwise adjust the `make_snap_installable` helper